src/collision.cpp

```diff
@@ -72,6 +80,42 @@ inline v3f truncate(const v3f vec, const f32 factor)
 	);
 }
 
+inline v3f rangelimv(const v3f vec, const f32 low, const f32 high)
+{
+	return v3f(
+		rangelim(vec.X, low, high),
+		rangelim(vec.Y, low, high),
+		rangelim(vec.Z, low, high)
+	);
+}
+}
+
+const f32 NO_COLLISION_TIME = -std::numeric_limits<f32>::infinity();
+
+// Time to collision in a one dimensional setting, with acceleration.
+// We have `pos = vel * dtime + 0.5 * acc * dtime²`.
+// This yields `dtime = (-vel ± sqrt(vel² + 2 * acc * pos)) / acc`
+inline f32 timeToCollision1D(f32 pos, f32 vel, f32 acc) {
```

Comment:
```suggestion
inline f32 time_to_collision_1d(f32 pos, f32 vel, f32 acc)
{
```
According to the code style guidelines.

EDIT: Interestingly, `inline` produces an 48 bytes smaller binary than with `static` (or no keyword) on GCC 11.4.0 (Release build).